src/App.tsx

```diff
@@ -15,49 +20,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface SortParams<T> {
+  sortReverse: boolean;
+  sortField: keyof T | '';
+}
+
+function getPreparedGoods<T>(
+  goods: T[],
+  { sortReverse, sortField }: SortParams<T>,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((element1, element2) => {
+      const value1 = element1[sortField];
+      const value2 = element2[sortField];
+
+      if (typeof value1 === 'number' && typeof value2 === 'number') {
+        return value1 - value2;
+      }
+
+      if (typeof value1 === 'string' && typeof value2 === 'string') {
+        return value1.localeCompare(value2);
+      }
+
+      return 0;
+    });
+  }
+
+  if (sortReverse) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<keyof SortType | ''>('');
+  const [sortReverse, setSortReverse] = useState(false);
```

Comment:
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```